cfn-resources/cluster/cmd/resource/mappings.go

```diff
@@ -39,6 +39,7 @@ func mapClusterToModel(model *Model, cluster *admin.AdvancedClusterDescription)
 	model.CreatedDate = util.TimePtrToStringPtr(cluster.CreateDate)
 	model.DiskSizeGB = cluster.DiskSizeGB
 	model.EncryptionAtRestProvider = cluster.EncryptionAtRestProvider
+	model.GlobalClusterSelfManagedSharding = cluster.GlobalClusterSelfManagedSharding
```

Comment:
i have mixed feelings here. On the one hand I think it's a good idea to have contract tests about as many features and resources as possible.
on the other hand this feature requires GEOSHARDED clusters that are very slow to create, update, etc., and we'll be probably testing more Atlas than the CFN logic, so in this case I'm not sure about the pros/cons balance

Reply:
I understand but I'm not sure if that's a good enough reason to not have a test at all. Looking at the current [inputs](https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/cfn-resources/cluster/test/inputs_1_create.template.json) file for cluster CFN resource, it looks like there is very minimal coverage as it is and only one REPLICASET test case with only one replication spec is being tested.

Also I don't see cluster tests running in CI.

I'd suggest we should have a test to avoid further divergence from the resource. Since the test doesn't run in CI anyway it would be good to at least have the option to run it locally.

Reply:
if there are not e2e or contract tests cluster tests we should definitely have.

my point was that we should have only some basic tests in e2e / contract tests, not all different combinations of parameters and features, that can normally better tested in unit tests.

in this case I'm not sure about having a specific e2e or contract test only to test this attribute, but I agree to have some e2e and contract tests running in CI for cluster.

Reply:
Added a follow-up ticket [here](https://jira.mongodb.org/browse/CLOUDP-256730)

Reply:
>not all different combinations of parameters and features, that can be normally better tested in unit tests.

I'm not suggesting to have tests for all combinations of parameters, but at least for different types of cluster, I would consider that basic coverage. 
In this case, the type would be GEOSHARDED using this attribute.